blocks/library/tag-cloud/block.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	withAPIData,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TermList from './term-list';
+import InspectorControls from '../../inspector-controls';
+import BlockControls from '../../block-controls';
+import BlockAlignmentToolbar from '../../block-alignment-toolbar';
+
+class TagCloudBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomies() {
+		const taxonomies = this.props.taxonomies.data;
+
+		if ( ! taxonomies ) {
```

Comment:
Minor: Lodash's `_.map`, which you're using, will gracefully handle `undefined` values:

```
n_ > _.map( undefined, () => '' );
[]
```